lib/puppet/type/zfs.rb

```diff
@@ -56,6 +56,15 @@ module Puppet
       desc 'The dedup property. Valid values are `on`, `off`.'
     end
 
+    newproperty(:defaultuserquota) do
+      desc 'The defaultuserquota property. Valid values are `<size>`, `none`.'
+
+      # Solaris 10 uses zfs filesystem version 5 which doesn't support defaultuserquota
+      validate do |_value|
+        raise Puppet::Error _('This property is only supported on zfs filestystem version >= 6') if Facter.value(:kernel) == 'SunOS' && Facter.value('os.release.major') == '10'
```

Comment:
I'm not sure if `Facter.value('os.release.major')` is best practice? It will work in facter 4 but not facter 3, similar to https://github.com/puppetlabs/puppet-enterprise-modules/pull/2008